ERCS/erc-7621.md

```diff
@@ -0,0 +1,140 @@
+
+#### Burning the BTS LP tokens
+
+The burning of BTS LP tokens occurs during the withdraw function. Users can call this function, specifying the amount of BTS LP tokens they want to withdraw. The contract then transfers the specified amount of BTS LP tokens from the user to the BTS pair contract (btsPair). The burn function of the IBTSPair contract is called, which decreases the user's BTS LP token balance and returns an array of amounts representing the underlying tokens withdrawn. This array of amounts is logged through the WithdrawnFromBTS event.
+
+#### Events
+
+ContributedToBTS(address indexed bts, uint256 amount): Event when someone adds ETH to the basket.
+
+WithdrawnFromBTS(address indexed bts, uint256[] amounts): Event when BTS LP holder withdraws tokens from the basket.
+
+WithdrawnETHFromBTS(address indexed bts, uint256 amount): Event when BTS LP holder withdraws ETH from the basket.
+
+RebalanceBTS(address indexed bts, uint256[] oldWeights, uint256[] newWeights): Event when the owner manually rebalances the basket.
+
+#### Storage
+
+btsPair: A public variable storing the address of the associated BTS pair contract.
+
+autoRebalanceEnabled: A public variable indicating whether auto-rebalancing is enabled or not.
+
+mapping(address => bool) private _isTokenPresent: A private mapping indicating whether a token is present or not.
+
+TokenDetails: A structure holding details about the tokens, including arrays for token addresses and their corresponding weights.
+
+_tokenDetails: An instance of the TokenDetails structure.
+
+### BTSPair
+
+#### Purpose
+
+[ERC-20](./eip-20.md) token representing liquidity in a BTS.
+
+#### Key Functions
+
+initialize: Initializes a new BTSPair with a name and tokens.
+
+rebalance(): Allows the owner to manually adjust the types and weights of tokens in the pair.
+
+updateTokens(...): Allows the owner to change the types of tokens in the pair.
+
+mint(...): Creates new BTS LP tokens and adjusts token reserves.
+
+burn(...): Destroys BTS LP tokens and adjusts token reserves.
+
+#### Storage
+
+tokens: Array of token addresses in the pair.
+
+reserves: Array of corresponding token reserves.
+
+### IUniswap
+
+#### Purpose
+
+Defines interfaces for interacting with Uniswap V2 Router and Factory contracts (external dependencies).
+
+## Rationale
+
+TBD
+
+## Backwards Compatibility
+
+No backward compatibility issues found.
+
+## Reference Implementation
+
+BTS.sol: Implemented using a combination of Solidity's built-in [ERC-721](./eip-721.md)URIStorage and Ownable contracts, along with custom logic for token management and rebalancing.
+
+Factory.sol: Implemented using ClonesUpgradeable to deploy new BTS and BTSPair contracts.
+
+BTSPair.sol: Implemented using [ERC-20](./eip-20.md)Upgradeable and Ownable contracts, along with custom logic for liquidity tracking and rebalancing.
+
+IUniswap.sol: Not implemented, as it only defines interfaces for external contracts.
+
+## Security Considerations
```

Comment:
If you include an HTML style comment like `<!-- TODO -->`, the linter will remind you to fill in the section.